### DISCUSSION TITLE
Notification browsing mode and search filtering

### DISCUSSION THREAD

**Discussion by Raivshard** [civitai/civitai#331] [Ideas]
I suggest creating a notification mode selector (like browsing mode) and additionally adding live filtering options in a separate search box to the notification list. 
 
**Live filtering**: A search box that filters in or out what you type
- ony filter the list after a short delay so that it doesn't instantly try to filter out half-finished words and spend unnecessary processing power)
- drops down tag suggestions as you type so you can select them (ie. filter out #[NSFW] or #[genshin impact], etc) or just type whatever and have that be filtered on your list
- include a switch for positive or negative filtering.
- maybe include 

Currently, adding filters to your notifications is done via settings. You have to go back into settings and change it again if you want it off. This is klunky and you have to go back and forth to see changes, which costs time.

**Notification Mode**:
- take the gearbox Notification Settings button and turn it into a popup selector just like the Browsing Mode popup
- Just like the Browsing Mode popup on the main site, but with some additional options for all the different types of notifications. 
- Basically, take the list of notifications settings from the settings page and add them to the box.
- link the popup and the settings page, have them share 

And while we're at it, how about saving the scrollbar position within the same session, or, say, the past X minutes? That way when you click on something, grab a new file, then quickly go back, you don't have to search through the 200 notifications that came through since you last checked.
